score-lang-compiler/src/main/java/org/openscore/lang/compiler/SlangCompilerImpl.java

```diff
@@ -181,45 +170,32 @@ private ExecutionPlan compileToExecutionPlan(Executable executable) {
      * also handles operations files
      *
      * @param parsedSlang the source to transform
-     * @return List of {@link org.openscore.lang.compiler.model.Executable}  of the requested flow/operations
+     * @return List of {@link org.openscore.lang.compiler.model.Executable}  of the requested flow or operation
      */
-    private List<Executable> transformToExecutables(ParsedSlang parsedSlang) {
-        List<Executable> executables;
-
+    private Executable transformToExecutable(ParsedSlang parsedSlang) {
         switch (parsedSlang.getType()) {
-            case OPERATIONS:
-                executables = transformOperations(parsedSlang);
-                break;
+            case OPERATION:
+                return transformOperation(parsedSlang);
             case FLOW:
-                executables = new ArrayList<>();
-                executables.add(transformFlow(parsedSlang));
-                break;
+                return transformFlow(parsedSlang);
             default:
                 throw new RuntimeException("source: " + parsedSlang.getName() + " is not of flow type or operations");
         }
-        return executables;
     }
 
     /**
-     * transform an operations {@link org.openscore.lang.compiler.model.ParsedSlang} to a List of {@link org.openscore.lang.compiler.model.Executable}
+     * transform an operation {@link org.openscore.lang.compiler.model.ParsedSlang} to a List of {@link org.openscore.lang.compiler.model.Executable}
      *
      * @param parsedSlang the source to transform the operations from
-     * @return List of {@link org.openscore.lang.compiler.model.Executable} representing the operations in the source
+     * @return {@link org.openscore.lang.compiler.model.Executable} representing the operation in the source
      */
-    private List<Executable> transformOperations(ParsedSlang parsedSlang) {
-        List<Executable> executables = new ArrayList<>();
-        for (Map<String, Map<String, Object>> operation : parsedSlang.getOperations()) {
-            Map.Entry<String, Map<String, Object>> entry = operation.entrySet().iterator().next();
-            String operationName = entry.getKey();
-            Map<String, Object> operationRawData;
-            try {
-                operationRawData = entry.getValue();
-            } catch (ClassCastException ex){
-                throw new RuntimeException("Operation: " + operationName + " syntax is illegal. Below operation name, there should be a map of values.");
-            }
-            executables.add(executableBuilder.transformToExecutable(parsedSlang, operationName, operationRawData));
+    private Executable transformOperation(ParsedSlang parsedSlang) {
```

Comment:
I prefer that we will have one method for both flow and operation and than we will switch case only once